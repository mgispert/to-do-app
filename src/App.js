import { useState } from "react";
import { ChakraProvider } from "@chakra-ui/react";
import { TaskBar } from "./components/TaskBar";
import { TaskList } from "./components/TaskList";

export const App = () => {
  const [toDoTasks, setToDoTasks] = useState([]);
  const [doneTasks, setDoneTasks] = useState([]);

  const handleOnAddTask = (valueInput) =>
    setToDoTasks((oldToDoTasks) => [valueInput, ...oldToDoTasks]);

  const handleOnClickToDoTask = (taskToMove) => {
    const excludeDoneTaskFromToDoTasks = toDoTasks.filter(
      (toDoTask) => toDoTask !== taskToMove
    );
    setToDoTasks(excludeDoneTaskFromToDoTasks);
    setDoneTasks((oldDoneTasks) => [taskToMove, ...oldDoneTasks]);
  };

  return (
    <ChakraProvider>
      <h1>To Do App</h1>
      <TaskBar onAddTask={handleOnAddTask} />
      <TaskList
        toDoTasks={toDoTasks}
        doneTasks={doneTasks}
        onClickToDoTask={handleOnClickToDoTask}
      />
    </ChakraProvider>
  );
};
