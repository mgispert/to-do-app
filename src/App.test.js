import { render, screen, fireEvent } from "@testing-library/react";
import { App } from "./App";

describe("App", () => {
  const selectAndChangeInput = () => {
    const input = screen.getByPlaceholderText(/add your task here/i);
    fireEvent.change(input, { target: { value: "clean" } });
  };

  const selectAndClickButton = () => {
    const button = screen.getByText(/add task/i);
    fireEvent.click(button);
  };

  const selectToDoVisibleTask = (shouldBeClicked) => {
    const toDoTask = screen.getByText(/clean/i);
    expect(toDoTask).toBeVisible();
    shouldBeClicked && fireEvent.click(toDoTask);
  };

  it("should match snapshot", () => {
    const { container } = render(<App />);
    expect(container).toMatchSnapshot();
  });
  it("should add a to do task and render it to the list", () => {
    render(<App />);
    selectAndChangeInput();
    selectAndClickButton();
    selectToDoVisibleTask(false);
  });
  it("should move a to do task towards done task list when clicked", () => {
    render(<App />);
    selectAndChangeInput();
    selectAndClickButton();
    selectToDoVisibleTask(true);
    expect(screen.getByText(/clean/i)).not.toBeVisible();
    const doneTab = screen.getByText(/done/i);
    fireEvent.click(doneTab);
    selectToDoVisibleTask(false);
  });
});
