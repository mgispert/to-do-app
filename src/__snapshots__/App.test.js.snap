// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App should match snapshot 1`] = `
<div>
  <h1>
    To Do App
  </h1>
  <form
    data-testid="taskBar"
    id="text"
  >
    <label
      class="chakra-form__label css-2gx1h6"
    >
      Write a task to add!
    </label>
    <input
      class="chakra-input css-1c6j008"
      placeholder="Add your task here"
      type="text"
      value=""
    />
    <button
      class="chakra-button css-taj3dd"
      type="submit"
    >
      Add Task
    </button>
  </form>
  <div
    class="chakra-tabs css-13o7eu2"
    data-testid="taskList"
  >
    <div
      aria-orientation="horizontal"
      class="chakra-tabs__tablist css-1xhq01z"
      role="tablist"
    >
      <button
        aria-controls="tabs-1--tabpanel-0"
        aria-selected="true"
        class="chakra-tabs__tab css-1ltezim"
        data-index="0"
        id="tabs-1--tab-0"
        role="tab"
        tabindex="0"
        type="button"
      >
        To Do
      </button>
      <button
        aria-controls="tabs-1--tabpanel-1"
        aria-selected="false"
        class="chakra-tabs__tab css-1ltezim"
        data-index="1"
        id="tabs-1--tab-1"
        role="tab"
        tabindex="-1"
        type="button"
      >
        Done
      </button>
    </div>
    <div
      class="chakra-tabs__tab-panels css-8atqhb"
    >
      <div
        aria-labelledby="tabs-1--tab-0"
        class="chakra-tabs__tab-panel css-a5mhaz"
        id="tabs-1--tabpanel-0"
        role="tabpanel"
        tabindex="0"
      >
        <ul
          class="css-1xa84ef"
          role="list"
        />
      </div>
      <div
        aria-labelledby="tabs-1--tab-1"
        class="chakra-tabs__tab-panel css-a5mhaz"
        hidden=""
        id="tabs-1--tabpanel-1"
        role="tabpanel"
        tabindex="0"
      >
        <ul
          class="css-1xa84ef"
          role="list"
        />
      </div>
    </div>
  </div>
</div>
`;
